pkg/controller/util.go

```diff
@@ -118,16 +119,12 @@ var emptyName = (spec.NamespacedName{})
 // configuration in ConfigMap & CRD.
 func (c *Controller) getInfrastructureRoleDefinitions() []*config.InfrastructureRole {
 	var roleDef config.InfrastructureRole
-	rolesDefs := c.opConfig.InfrastructureRoles
 
-	if c.opConfig.InfrastructureRolesSecretName == emptyName {
-		// All the other possibilities require secret name to be present, so if
-		// it is not, then nothing else to be done here.
-		return rolesDefs
-	}
+	// take from CRD configuration
+	rolesDefs := c.opConfig.InfrastructureRoles
 
 	// check if we can extract something from the configmap config option
-	if c.opConfig.InfrastructureRolesDefs != "" {
+	if len(rolesDefs) == 0 && c.opConfig.InfrastructureRolesDefs != "" {
```

Comment:
No one prevents having both new roles definitions and old format, AFAICS that was the idea.